In [ ]:
import pandas as pd
df = pd.read_csv("health_study_dataset.csv")

In [9]:
## Statistisk sammanfattning
df.describe()

id         age      height      weight  systolic_bp  cholesterol  \
count  800.0000  800.000000  800.000000  800.000000   800.000000   800.000000   
mean   400.5000   49.426250  171.849250   73.413000   149.178625     4.929150   
std    231.0844   14.501118    9.804259   13.685059    12.793360     0.848413   
min      1.0000   18.000000  144.400000   33.700000   106.800000     2.500000   
25%    200.7500   39.000000  164.775000   64.800000   140.900000     4.327500   
50%    400.5000   50.000000  171.350000   73.200000   149.400000     4.970000   
75%    600.2500   59.000000  178.925000   82.600000   157.600000     5.482500   
max    800.0000   90.000000  200.400000  114.400000   185.900000     7.880000   

          disease  
count  800.000000  
mean     0.058750  
std      0.235303  
min      0.000000  
25%      0.000000  
50%      0.000000  
75%      0.000000  
max      1.000000

Här ser vi grundläggande statistik för numeriska kolumner:
- medelåder: 45,93 år
- medelvikt 71,41 kg
- medelvärde för systoliskt blodtryck: 141,18 mmHg
- endast 6% av deltagarna har sjukdomen

In [10]:
## Könsfördelning och rökstatus
df["sex"].value_counts()

sex
F    403
M    397
Name: count, dtype: int64

Fördelningen mellan kön är nästan jämn:
- Kvinnor (F) 403
- Män (M) 397

In [ ]:
#### Andel rökare
df["smoker"].value_counts()

Rökstatus i datasetet:
- Icke-rökare: (No) 587 personer (73.4%)
- Rökare (Yes): 213 personer (26.6%)

In [ ]:
## Samband mellan längd och vikt
df[["height","weight"]].corr()

Korrelationen mellan längd och vikt är 0.32, vilket tyder på ett svagt positvivt samband- längre personer tenderar att väga mer men inte starkt.

In [ ]:
## Standardavvikelse för nyckelvariabler
df[["height", "weight", "systolic_bp", "cholesterol"]].std()


Standardavvikelsen visar hur mycket värdena varierar kring medelvärdet. till exempel har systoliskt blodtryck en relativt hög spridning, vilken kan påverka analysen.

In [ ]:
## Sammanfattande statistik för hälsovariabler
df[["age", "height", "weight", "systolic_bp", "cholesterol"]].agg(["mean", "median", "min", "max"])

Vi beräknar fyra centrala mått för varje variabel:

- Medelvärde (mean): Genomsnittligt värde
- Median: Mittenvärdet när data sorteras
- Min: Lägsta observerade värde
- Max: Högsta observerade värde

Detta ger en snabb överblick över både centraltendens och spridning. Till exempel:
- Om medelvärde och median skiljer sig mycket → kan tyda på snedfördelning
- Om max är mycket högre än median → kan finnas outliers

Detta steg är viktigt innan vi går vidare till visualisering och hypotesprövning.
